Userland/Libraries/LibWeb/HTML/Scripting/Fetching.cpp

```diff
@@ -423,6 +426,83 @@ WebIDL::ExceptionOr<void> fetch_classic_worker_script(URL::URL const& url, Envir
     return {};
 }
 
+// https://html.spec.whatwg.org/multipage/webappapis.html#fetch-a-classic-worker-imported-script
+WebIDL::ExceptionOr<JS::NonnullGCPtr<ClassicScript>> fetch_a_classic_worker_imported_script(URL::URL const& url, HTML::EnvironmentSettingsObject& settings_object, PerformTheFetchHook perform_fetch)
+{
+    auto& realm = settings_object.realm();
+    auto& vm = realm.vm();
+
+    // 1. Let response be null.
+    JS::GCPtr<Fetch::Infrastructure::Response> response = nullptr;
+
+    // 2. Let bodyBytes be null.
+    Fetch::Infrastructure::FetchAlgorithms::BodyBytes body_bytes;
+
+    // 3. Let request be a new request whose URL is url, client is settingsObject, destination is "script", initiator type is "other",
+    //    parser metadata is "not parser-inserted", and whose use-URL-credentials flag is set.
+    auto request = Fetch::Infrastructure::Request::create(vm);
+    request->set_url(url);
+    request->set_client(&settings_object);
+    request->set_destination(Fetch::Infrastructure::Request::Destination::Script);
+    request->set_initiator_type(Fetch::Infrastructure::Request::InitiatorType::Other);
+    request->set_parser_metadata(Fetch::Infrastructure::Request::ParserMetadata::NotParserInserted);
+    request->set_use_url_credentials(true);
+
+    auto process_response_consume_body = [&response, &body_bytes](JS::NonnullGCPtr<Fetch::Infrastructure::Response> res, Fetch::Infrastructure::FetchAlgorithms::BodyBytes bb) {
+        // 1. Set bodyBytes to bb.
+        body_bytes = move(bb);
+
+        // 2. Set response to res.
+        response = res;
+    };
+
+    // 4. If performFetch was given, run performFetch with request, isTopLevel, and with processResponseConsumeBody as defined below.
+    if (perform_fetch) {
+        TRY(perform_fetch->function()(request, TopLevelModule::Yes, move(process_response_consume_body)));
+    }
+    // Otherwise, fetch request with processResponseConsumeBody set to processResponseConsumeBody as defined below.
+    else {
+        Fetch::Infrastructure::FetchAlgorithms::Input fetch_algorithms_input {};
+        fetch_algorithms_input.process_response_consume_body = move(process_response_consume_body);
+        TRY(Fetch::Fetching::fetch(realm, request, Fetch::Infrastructure::FetchAlgorithms::create(vm, move(fetch_algorithms_input))));
+    }
+
+    // 5. Pause until response is not null.
+    auto& event_loop = settings_object.responsible_event_loop();
+    event_loop.spin_until([&]() {
+        return response;
+    });
```

Comment:
This spins forever most of the time when waiting for a file being fetched over HTTP. When fetching local files everything works as expected.

Is there a way I can pause here that will play nicely with our fetch implementation?